File: src/OpenDHT/Blob.c

#include <opendht/opendht_c.h>

void dht_blob_get_data_view(dht_data_view* data_view, const dht_blob* data) {
    dht_data_view data_view_ = dht_blob_get_data(data);
    data_view->data = data_view_.data;
    data_view->size = data_view_.size;
};

/*  vim: set sts=4 ts=4 sw=4 tw=120 et : */

File: src/OpenDHT/Blob.hs

{-|
  Module      : OpenDHT.Blob
  Description : Blob related bindings
  Copyright   : (c) Simon Désaulniers, 2025
  License     : GPL-3

  Maintainer  : sim.desaulniers@gmail.com
-}

module OpenDHT.Blob ( Blob
                    ) where

import qualified Data.ByteString as BS

{-| This data type holds a pointer to data to be stored or retrieved on the DHT.

  In fact, in OpenDHT, "Blob" is a typdef to "std::vector<uint8_t>".
-}
type Blob = BS.ByteString

--  vim: set sts=2 ts=2 sw=2 tw=120 et :

